httpx/models.py

```diff
@@ -93,7 +94,11 @@ def __init__(
 
         # Handle IDNA domain names.
         if self.components.authority:
-            idna_authority = self.components.authority.encode("idna").decode("ascii")
+            # idna.encode raises InvalidCodepoint when encountering a colon, so split
+            # host and port in case the latter is specified and rejoin after encoding
+            host_port = self.components.authority.split(":")
```

Comment:
I think there's a way we can rely on rfc3986 to hand us the host portion of a URI. The current implementation would choke on IPv6 addresses.

Reply:
This is true, I had also assumed that authority would always be a fqdn/hostname. Getting just the host portion to start with would be best.